pandas/io/packers.py

```diff
@@ -595,11 +595,14 @@ def decode(obj):
         pd_dtype = pandas_dtype(dtype)
         np_dtype = pandas_dtype(dtype).base
 
+        # see gh-14901
+        klass_dtype = np_dtype if pd_dtype.name != 'category' else pd_dtype
+
         index = obj[u'index']
         result = globals()[obj[u'klass']](unconvert(obj[u'data'], dtype,
                                                     obj[u'compress']),
                                           index=index,
-                                          dtype=np_dtype,
+                                          dtype=klass_dtype,
```

Comment:
`s.get_values()` is what you get when you call `unconvert` in `read_msgpack`.  `s2` is what you create when you call `globals()[obj[u'klass']](...)` if you pass in `dtype=s.dtype` (and not `s.dtype.base`, which is what we do now).